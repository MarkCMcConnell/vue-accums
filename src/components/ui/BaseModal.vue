<template>
    <div class="modalBackdrop">
        <div class="modal-container">
            <header class="modal-header">
                    <slot name="modalTitle"></slot>
            </header>

            <section  class="modal-body">
                <slot name="modalBody"/>
            </section>
            <footer class="modal-footer">
                <slot name="modalFooter" />
                <button
                    class="modal-button btn-primary"
                    @click.prevent="HIDE_MODAL"
                >
                    <slot name="modalButton">Close</slot>
                </button>
            </footer>
        </div>
    </div>
</template>

<script>
import { mapMutations } from 'vuex'

export default {
    methods: {
        ...mapMutations([
            'HIDE_MODAL'
        ])
    }
}
</script>

<style scoped>
    .modalBackdrop {
        position: fixed;
        top: 0;
        bottom: 0;
        left: 0;
        right: 0;
        background-color: rgba(0, 0, 0, 0.5);
        display: flex;
        justify-content: center;
        align-items: center;
    }

    .modal-container {
        width: 50%;
        position: absolute;
        top: 50%;
        left: 50%;
        transform: translate( -50%, -50%);
        background-color: #fff;
    }
</style>
