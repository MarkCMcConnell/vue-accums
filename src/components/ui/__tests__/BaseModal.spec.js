import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'

import BaseModal from '@/components/ui/BaseModal.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('Modal component', () => {
  describe('Modal rendering', () => {
    let wrapper

    beforeEach(() => {
      wrapper = shallowMount(BaseModal, {
        slots: {
          modalHeader: '<h2>Title</h2>',
          modalBody: '<p>Some Content</p>',
          modalFooter: '<p>More Content</p>',
          modalButton: 'Close'
        }
      })
    })

    it('renders correctly', () => {
      expect(wrapper.find('.modalBackdrop')).toBeTruthy()
    })

    it('renders slot content', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('Modal behavior', () => {
    let wrapper
    let store
    let mutations

    beforeEach(() => {
      mutations = {
        HIDE_MODAL: jest.fn()
      }
      store = new Vuex.Store({
        modules: {
          modal: {
            state: {
              showModal: true
            },
            mutations
          }
        }
      })

      wrapper = shallowMount(BaseModal, {
        store,
        localVue
      })
    })

    it('commits a HIDE_MODAL mutation when clicked', () => {
      wrapper.find('.modal-button').trigger('click')
      expect(mutations.HIDE_MODAL).toHaveBeenCalled()
    })
  })
})
