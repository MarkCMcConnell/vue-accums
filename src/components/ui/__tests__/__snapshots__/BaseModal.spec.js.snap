// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component Modal rendering renders slot content 1`] = `
<div class="modalBackdrop">
  <div class="modal-container">
    <header class="modal-header"></header>
    <section class="modal-body">
      <p>Some Content</p>
    </section>
    <footer class="modal-footer">
      <p>More Content</p> <button class="modal-button btn-primary">Close</button>
    </footer>
  </div>
</div>
`;
